src/app/_components/EventLinkModal.js

```diff
@@ -1,13 +1,55 @@
 import PropTypes from 'prop-types';
 import { QRCodeSVG } from 'qrcode.react';
+import Instagram from '@icons/instagram.svg';
+import Facebook from '@icons/facebook.svg';
 import { Modal } from '@components/Modal';
 import { PillButton } from '@components/PillButton';
 import { Heading, Paragraph } from '@components/Typography';
+import { useMediaQuery } from '@hooks';
+import { screens } from '@/app/styles/tailwind/ui';
+
+function MobileModal({ link }) {
+  // refactor this icon to use 'let' and two if's to make it more readable
+  let icon = null;
+  if (link.includes('instagram')) {
+    icon = <Instagram />;
+  }
+  if (link.includes('facebook')) {
+    icon = <Facebook />;
+  }
+
+  return (
+    <a className="lg:hidden" href={link} target="_blank" rel="noopener noreferrer">
+      <PillButton
+        variant="filled"
+        colorVariant="blue"
+        className="flex w-full text-center font-bold text-other-white"
+        aria-label="Перейти за лінком"
+      >
+        {icon ? (
+          <span className="flex gap-4">
+            <span>Відкрити в</span>
+            {icon}
+          </span>
+        ) : (
+          <span>Перейти за лінком</span>
+        )}
+      </PillButton>
+    </a>
+  );
+}
 
 export function EventLinkModal({ isOpen, onClose, link }) {
+  const isDesktop = useMediaQuery(`(min-width: ${screens.lg})`);
+
   return (
-    <Modal isOpen={isOpen} onClose={onClose} className="bg-primary-200">
-      <div className="flex flex-col items-center gap-6 px-[72px] pb-12 pt-3 lg:w-[744px]">
+    <Modal
+      isOpen={isOpen}
+      onClose={onClose}
+      className="w-64 lg:bg-primary-200 opacity-100 lg:w-fit"
+      isCloseButton={isDesktop}
```

Comment:
you can check if the behavior with the hiding close button if not desktop has to be for all modal windows in app you can move this logic inside the component